### PR TITLE
Fix form and location syncing bugs

### PR DESCRIPTION
**Pull request by georific**
## Status
**READY**

## JIRA Ticket
SPACON-640

## Description
Fixed location to sending properly, location being added to form submissions, stopped duplicates

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

```sh
git fetch --all
git checkout <feature_branch> 
./gradlew connectedCheck
```

@boundlessgeo/spatial-connect

### CODE REVIEW

**Comment by frankrowe** [comment]
👍

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/geometries/SCSpatialFeature.java [hunk 1]
```diff
@@ -130,4 +130,16 @@ public String toJson() {
...
         return null;
     }
+
+    public  Map<String, Object> toMap() {
+        Map<String, Object> map = null;
+        try {
+            map = SCObjectMapper.getMapper().convertValue(this,  Map.class);
+            map.put("type", "Feature");
+        } catch (Exception e) {
+            String s = "";
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/geometries/SCSpatialFeature.java]: Can you log the exception as a warning here?  Also, `String s = "";` doesn't appear to be doing anything

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/LocationStore.java [hunk 2]
```diff
@@ -110,25 +109,39 @@ public String syncChannel() {
...
         ss.isConnected().subscribe(new Action1<Boolean>() {
             @Override
             public void call(Boolean connected) {
                 if (connected) {
-                    SpatialConnect sc = SpatialConnect.getInstance();
-                    if (sc.getSensorService() != null) {
-                        SpatialConnect.getInstance()
-                                .getSensorService()
-                                .getLastKnownLocation()
+//                    SpatialConnect sc = SpatialConnect.getInstance();
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/LocationStore.java]: You can delete this line

**Comment by mrcnc** [comment]
👍

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/SCBackendService.java [hunk 3]
```diff
@@ -536,47 +535,46 @@ private Timestamp getTimestamp() {
...
         int sdkVersion = Build.VERSION.SDK_INT;
-        return "Android SDK: " + sdkVersion + " (" + release + ")";
+        String os =  String.format("Android SDK: %s (%s)", sdkVersion, release);
+        return String.format("{\"os\": \"%s\"}", os);
     }
 
-    private void setupSyncListener() {
+    private void listenForSyncEvents() {
+
+        Observable<SCDataStore> syncableStores = dataService.getISCSpatialStores();
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/SCBackendService.java]: Should this be `dataService.getISyncableStores()`?

**Comment by mrcnc** [comment]
👍

**Pull request merged**